File: vol2/ch16/LeetCode344.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        rev(s,0,s.size()-1);
    }
    void rev(vector<char>&s,int i,int j) {
        if(i<j) {       											//至少含两个字符
            swap(s[i],s[j]);
            rev(s,i+1,j-1);
        }
    }
};

File: vol2/ch16/LeetCode344.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        self.rev(s,0,len(s)-1)
    def rev(self,s,i,j):
        if i<j:       #至少含两个字符
            s[i],s[j]=s[j],s[i]
            self.rev(s,i+1,j-1)
